cogs/ctftime.py

```diff
@@ -0,0 +1,108 @@
+from datetime import datetime
+
+import discord
+import requests
+from discord.ext import commands
+
+from .utils.ctftimeapiutils import fetch_event_details, fetch_upcoming_events
+
+
+class CFTTime(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.group()
+    async def ctftime(self, ctx):
+        # Default command that shows a list of subcommands
+        if ctx.invoked_subcommand is None:
+            embed = discord.Embed(title="CTFTime Commands", color=discord.Color.blue())
+            embed.add_field(
+                name="upcoming", value="Show upcoming CTF events.", inline=False
+            )
+            embed.add_field(
+                name="details [event_id]",
+                value="Show details of a CTF event with the specified ID.",
+                inline=False,
+            )
+            await ctx.send(embed=embed)
```

Comment:
Let's change this to a [slash command group](https://guide.pycord.dev/interactions/application-commands/slash-commands#subcommand-groups). That'll make the command easier to use and nolonger require a help function
```suggestion
    ctftime = SlashCommandGroup("ctftime", "CTFTime commands")
```